source/server/server.h

```diff
@@ -141,6 +141,35 @@ class RunHelper : Logger::Loggable<Logger::Id::main> {
   Event::SignalEventPtr sig_hup_;
 };
 
+class ServerFactoryContextImpl : public Configuration::ServerFactoryContext {
```

Comment:
So, the way I envisaged this playing out is that the `ListenerImpl` factory context should now be massively slimmed down. It doesn't need to be offering most of the methods it has, they can come via the server factory context. But, I don't see a lot of change to listener manager in this PR.

Reply:
Sorry I don't fully get your idea.
Are you saying ListenerFactoryContext should exclude what ServerFactoryContext provides? Or are you suggesting ListenerImpl should offer the `ServerFactoryContext` functions through a impl of `ServerFactoryContext`?